libs/durability/disk_failure_recovery_libs.py

```diff
@@ -220,3 +234,113 @@ def fail_disk(self, disk_fail_cnt: int, master_obj: LogicalNode,
             failed_disks_dict['disk' + str(cnt)] = all_disks[selected_disk]
             all_disks.pop(selected_disk)
         return True, failed_disks_dict
+
+    def get_user_data_space_in_bytes(self, master_obj: LogicalNode, memory_percent: int) -> tuple:
+        """
+        Retrieve the available user data space in bytes to attain the given memory percent
+        :param master_obj: Logical node object of Master
+        :param memory_percent: Expected Memory Percentage to achieve.
+        :return : (boolean, int)
+        """
+        # Get available data disk space
+        health_obj = Health(master_obj.hostname, master_obj.username, master_obj.password)
+        total_cap, avail_cap, used_cap = health_obj.get_sys_capacity()
+        current_usage_per = int(used_cap / total_cap * 100)
+
+        if memory_percent > current_usage_per:
+            # Get SNS configuration to retrieve available user data space
+            durability_values = self.retrieve_durability_values(master_obj, 'sns')
+            if not durability_values[0]:
+                LOGGER.error("Error in retrieving SNS values")
+                return durability_values
```

Comment:
do we want to return false as well to assert in test?

Reply:
durability_values is itself tuple, hence False will be returned for test to assert.